In [1]:
 %load_ext sql

In [2]:
 %sql mysql+pymysql://root:mysql@localhost/sales

'Connected: root@sales'

In [11]:
%%sql
SELECT *
FROM orders_1
LIMIT 5;

 * mysql+pymysql://root:***@localhost/sales
5 rows affected.


[(10234, '2004-03-30', '2004-04-05', '2004-04-02', 'Shipped', 412, 'S72_1253', 40, 45690),
 (10234, '2004-03-30', '2004-04-05', '2004-04-02', 'Shipped', 412, 'S700_2047', 29, 83280),
 (10234, '2004-03-30', '2004-04-05', '2004-04-02', 'Shipped', 412, 'S24_3816', 31, 78830),
 (10234, '2004-03-30', '2004-04-05', '2004-04-02', 'Shipped', 412, 'S24_3420', 25, 65090),
 (10234, '2004-03-30', '2004-04-05', '2004-04-02', 'Shipped', 412, 'S24_2841', 44, 67140)]

In [12]:
%%sql
SELECT *
FROM orders_2
LIMIT 5;

 * mysql+pymysql://root:***@localhost/sales
5 rows affected.


[(10235, '2004-04-02', '2004-04-12', '2004-04-06', 'Shipped', 260, 'S18_2581', 24, 81950),
 (10235, '2004-04-02', '2004-04-12', '2004-04-06', 'Shipped', 260, 'S24_1785', 23, 89720),
 (10235, '2004-04-02', '2004-04-12', '2004-04-06', 'Shipped', 260, 'S24_3949', 33, 55270),
 (10235, '2004-04-02', '2004-04-12', '2004-04-06', 'Shipped', 260, 'S24_4278', 40, 63030),
 (10235, '2004-04-02', '2004-04-12', '2004-04-06', 'Shipped', 260, 'S32_1374', 41, 90900)]

In [10]:
%%sql
SELECT *
FROM customer
LIMIT 5;

 * mysql+pymysql://root:***@localhost/sales
5 rows affected.


[(103, 'Atelier graphique', 'Schmitt', 'Carine', 'Nantes', 'France', '2004-02-05'),
 (112, 'Signal Gift Stores', 'King', 'Jean', 'Las Vegas', 'USA', '2004-02-05'),
 (114, 'Australian Collectors, Co.', 'Ferguson', 'Peter', 'Melbourne', 'Australia', '2004-02-20'),
 (119, 'La Rochelle Gifts', 'Labrune', 'Janine', 'Nantes', 'France', '2004-02-05'),
 (121, 'Baane Mini Imports', 'Bergulfsen', 'Jonas', 'Stavern', 'Norway', '2004-02-05')]

# Total Penjualan dan Revenue pada Quarter-1 (Jan, Feb, Mar) dan Quarter-2 (Apr,Mei,Jun)

In [40]:
%%sql
# revenue Q1
SELECT SUM(quantity) AS total_penjualan,
       SUM(quantity * priceEach) AS revenue
FROM orders_1
WHERE status = 'Shipped';

 * mysql+pymysql://root:***@localhost/sales
1 rows affected.


[(Decimal('8694'), Decimal('799579310'))]

In [41]:
%%sql
# revenue Q2
SELECT SUM(quantity) AS total_penjualan,
       SUM(quantity * priceEach) AS revenue
FROM orders_2
WHERE status = 'Shipped';

 * mysql+pymysql://root:***@localhost/sales
1 rows affected.


[(Decimal('6717'), Decimal('607548320'))]

# Menghitung persentasi keseluruhan penjualan

In [34]:
%%sql
SELECT quarter,
       SUM(quantity) AS total_penjualan,
       SUM(quantity * priceEach) AS revenue
FROM (SELECT orderNumber,status, quantity, priceEach, 1 AS quarter
      FROM orders_1
      UNION
      SELECT orderNumber, status, quantity, priceEach, 2 AS quarter
      FROM orders_2) AS tabel_gabungan
WHERE status = 'Shipped'
GROUP BY quarter;

 * mysql+pymysql://root:***@localhost/sales
2 rows affected.


[(1, Decimal('8694'), Decimal('799579310')),
 (2, Decimal('6717'), Decimal('607548320'))]

# Perhitungan Growth Penjualan dan Revenue

# Apakah jumlah customers xyz.com semakin bertambah?

In [35]:
%%sql
SELECT quarter,
       COUNT(DISTINCT customerID) AS total_customers
FROM (SELECT customerID,
             createDate,
             QUARTER(createDate) AS quarter
      FROM customer
      WHERE createDate BETWEEN DATE('2004-01-01') AND DATE('2004-06-30')
      ) AS tabel_b
GROUP BY quarter;

 * mysql+pymysql://root:***@localhost/sales
2 rows affected.


[(1, 43), (2, 35)]

# Seberapa banyak customers tersebut yang sudah melakukan transaksi?

In [36]:
%%sql
SELECT quarter,
       COUNT(DISTINCT customerID) AS total_customers
FROM (SELECT customerID,
             createDate,
             QUARTER(createDate) AS quarter
      FROM customer
      WHERE (createDate BETWEEN DATE('2004-01-01') AND DATE('2004-06-30'))
      AND customerID IN
        (SELECT DISTINCT customerID
         FROM orders_1
         UNION SELECT DISTINCT customerID
         FROM orders_2)
      ) AS tabel_b
GROUP BY quarter;

 * mysql+pymysql://root:***@localhost/sales
2 rows affected.


[(1, 25), (2, 19)]

# Category produk apa saja yang paling banyak di-order oleh customers di Quarter-2?

In [42]:
%%sql
SELECT *
FROM (SELECT SUBSTR(productCode, 1, 3) AS categoryid,
      COUNT(DISTINCT orderNumber) AS total_order,
      SUM(quantity) AS total_penjualan
      FROM ( SELECT productCode,
                    orderNumber,
                    quantity,
                    status
             FROM orders_2
             WHERE
             status = 'Shipped'
            ) AS tabel_c
      GROUP BY categoryid
      ) AS tabel_a
ORDER BY total_order DESC;

 * mysql+pymysql://root:***@localhost/sales
8 rows affected.


[('S18', 25, Decimal('2264')),
 ('S24', 21, Decimal('1826')),
 ('S32', 11, Decimal('616')),
 ('S12', 10, Decimal('491')),
 ('S10', 8, Decimal('492')),
 ('S50', 8, Decimal('292')),
 ('S70', 7, Decimal('675')),
 ('S72', 2, Decimal('61'))]

# Seberapa banyak customers yang tetap aktif bertransaksi setelah transaksi pertamanya?

In [45]:
%%sql
SELECT COUNT(DISTINCT customerID) as total_customers
FROM orders_1;

 * mysql+pymysql://root:***@localhost/sales
1 rows affected.


[(25,)]

In [44]:
%%sql
SELECT 1 AS quarter,
       COUNT(DISTINCT customerID)*100/25 AS Q2
FROM orders_1
WHERE customerID IN (SELECT DISTINCT customerID
                     FROM orders_2)

 * mysql+pymysql://root:***@localhost/sales
1 rows affected.


[(1, Decimal('24.0000'))]

# Kesimpulan
Berdasarkan data yang telah kita peroleh melalui query SQL, Kita dapat menarik kesimpulan bahwa :
1. Performance xyz.com menurun signifikan di quarter ke-2, terlihat dari nilai penjualan dan revenue yang drop hingga 20% dan 24%,
2. Perolehan customer baru juga tidak terlalu baik, dan sedikit menurun dibandingkan quarter sebelumnya.
3. Ketertarikan customer baru untuk berbelanja di xyz.com masih kurang, hanya sekitar 56% saja yang sudah bertransaksi. Disarankan tim Produk untuk perlu mempelajari behaviour customer dan melakukan product improvement, sehingga conversion rate (register to transaction) dapat meningkat.
4. Produk kategori S18 dan S24 berkontribusi sekitar 50% dari total order dan 60% dari total penjualan, sehingga xyz.com sebaiknya fokus untuk pengembangan category S18 dan S24.
5. Retention rate customer xyz.com juga sangat rendah yaitu hanya 24%, artinya banyak customer yang sudah bertransaksi di quarter-1 tidak kembali melakukan order di quarter ke-2 (no repeat order).
6. xyz.com mengalami pertumbuhan negatif di quarter ke-2 dan perlu melakukan banyak improvement baik itu di sisi produk dan bisnis marketing, jika ingin mencapai target dan positif growth di quarter ke-3. Rendahnya retention rate dan conversion rate bisa menjadi diagnosa awal bahwa customer tidak tertarik/kurang puas/kecewa berbelanja di xyz.com.
